docs/methods/neighbors/categorical_nb/categorical_nb.md

```diff
@@ -0,0 +1,41 @@
+# Categorical Naive Bayes
+
+Categorical Naive Bayes model.
+
+Categorical Naive Bayes computes likelihood by : occurences of instance in current label divided by total occurences of instance
+
+It computes the prior probability by : occurence of current class divided by size of training array
+
+And finally prior*likelihood gives the probabilty of class according to which output class is predicted.
+
+
+## Parameters
+
+| Name   | Definition                               | Type               |
+|--------|------------------------------------------|--------------------|
+| `xTrain` | The training set containing the features | `vector<vector<T>>` |
+| `yTrain` | The set containing the class correspoding to respective xTrain instance | `vector<string`|
+| `xTest` | The sample set , whose class will be predicted | `vector<T>`|
+
+## Methods
+
+| Name                                    | Definition                                            | Return value |
+| -------------------------------         | ----------------------------------------------------- |--------------|
+| `fitPredict(vector<vector<T>>xTrain,vector<T>yTrain,vector<T>xTest)`        | fit and predict the training and testing values | `string`      |
```

Comment:
Fiex spacing between param name and param type. Also break it into two or more lines since there are many paramaters.

Reply:
sir , i have updated the readme wrt new methods